### PR TITLE
[chore] upgrade React devDeps to 15.6

### PR DESCRIPTION
**Pull request by ajfuller**
Fixes #[issue number].

Changes proposed:
- Upgrade React to the latest version
- Remove the deprecated `react-addons-test-utils` in favor of accessing from `react-dom/test-utils`
-

Upgrade Path (for changed or removed APIs):

Acceptance Checklist:
- [x] All commits have been squashed to one.
- [x] The commit message follows the guidelines in `CONTRIBUTING.md`.
- [x] Documentation (README.md) and examples have been updated as needed.
- [x] If this is a code change, a spec testing the functionality has been added.
- [x] If the commit message has [changed] or [removed], there is an upgrade path above.

### CODE REVIEW

File: specs/helper.js [hunk 1]
```diff
@@ -76,21 +75,32 @@ export const overlayAttribute = getModalAttribute('overlay');
...
-const dispatchMockEvent = eventCtor => (key, code) => (element, opts) =>
-  eventCtor(element, Object.assign({}, {
-    key: key, keyCode: code, which: code
-  }, opts));
+const dispatchMockEvent = eventCtor =>
+  (key, code) =>
+    (element, opts) =>
+      eventCtor(
+        element,
+        Object.assign(
```
Comment (diasbruno) [inline specs/helper.js]: Now, we can use `...` operator instead of `Object.assign`.

File: specs/helper.js [hunk 2]
```diff
@@ -76,21 +75,32 @@ export const overlayAttribute = getModalAttribute('overlay');
...
+          {},
+          {
+            key: key,
+            keyCode: code,
+            which: code
+          },
```
Comment (diasbruno) [inline specs/helper.js]: Now, we can use `{ key, keyCode, which: code }`.

Comment (diasbruno) [inline specs/helper.js]: ...or change the `code` on the arguments to `keyCode`. 
`{ key, keyCode, which: keyCode }`

**Comment by diasbruno** [comment]
@ajfuller I'll finish to update the changelog and I'll merge this PR, than we can release a version.

**Comment by ajfuller** [comment]
Sorry I have `prettier` enabled in Atom so it made a few more changes than I had expected. I'll make the requested changes.

**Comment by ajfuller** [comment]
@diasbruno I've cleaned this up so it's just the upgrade related changes. Other refactoring can be in a separate PR 😸.

**Pull request merged**